src/Agent.Plugins/PipelineCache/TarUtils.cs

```diff
@@ -198,16 +158,22 @@ private static void CreateProcessStartInfo(ProcessStartInfo processStartInfo, st
             processStartInfo.Arguments = processArguments;
             processStartInfo.UseShellExecute = false;
             processStartInfo.RedirectStandardInput = true;
-            processStartInfo.RedirectStandardOutput = true;
-            processStartInfo.RedirectStandardError = true;
             processStartInfo.WorkingDirectory = processWorkingDirectory;
         }
 
-        private static ProcessStartInfo GetCreateTarProcessInfo(string archiveFileName, string inputPath)
+        private static ProcessStartInfo GetCreateTarProcessInfo(AgentTaskPluginExecutionContext context, string archiveFileName, string inputPath)
         {
             var processFileName = "tar";
             inputPath = inputPath.TrimEnd(Path.DirectorySeparatorChar).TrimEnd(Path.AltDirectorySeparatorChar);
             var processArguments = $"-cf \"{archiveFileName}\" -C \"{inputPath}\" ."; // If given the absolute path for the '-cf' option, the GNU tar fails. The workaround is to start the tarring process in the temp directory, and simply speficy 'archive.tar' for that option.
+            if (context.Variables.TryGetValue("system.debug", out VariableValue systemDebugVar))
+            {
+                if (string.Equals(systemDebugVar?.Value, "true", StringComparison.OrdinalIgnoreCase))
+                {
+                    processArguments = $"-cvf \"{archiveFileName}\" -C \"{inputPath}\" .";
```

Comment:
I think this can be
if (....)
 processArguments = "-v " + processArguments

Reply:
oh right!